app/IATI/Traits/MigrateActivityPublishedTrait.php

```diff
@@ -0,0 +1,483 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\IATI\Traits;
+
+use App\Exceptions\PublishException;
+use App\IATI\Models\Activity\ActivityPublished;
+use App\IATI\Models\Organization\Organization;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+
+/**
+ * Class MigrateActivityPublishedTrait.
+ */
+trait MigrateActivityPublishedTrait
+{
+    /**
+     * Updates publisher id in organization level with that of settings.
+     *
+     * @param $iatiOrganization
+     * @param $aidStreamOrganizationSetting
+     * @param $aidStreamOrganization
+     *
+     * @return void
+     */
+    public function syncPublisherIdInSettingAndOrganizationLevel($iatiOrganization, $aidStreamOrganizationSetting, $aidStreamOrganization): void
+    {
+        $registryInfo = $aidStreamOrganizationSetting->registry_info;
+
+        if ($registryInfo) {
+            $registryInfo = json_decode($registryInfo)[0];
```

Comment:
Better to use Arr::get($registryInfo, 0 , []) so that system does not throw error if array does not have 0 key
Try practicing using Arr::get for  accessing array as it gives better error handling
Does not need to be implemented right away for this part but better to keep in mind